src/librbd/cache/pwl/AbstractWriteLog.cc

```diff
@@ -1461,10 +1461,10 @@ void AbstractWriteLog<I>::alloc_and_dispatch_io_req(C_BlockIORequestT *req)
 }
 
 template <typename I>
-bool AbstractWriteLog<I>::check_allocation(C_BlockIORequestT *req,
-      uint64_t &bytes_cached, uint64_t &bytes_dirtied, uint64_t &bytes_allocated,
-      uint64_t &num_lanes, uint64_t &num_log_entries,
-      uint64_t &num_unpublished_reserves) {
+bool AbstractWriteLog<I>::check_allocation(
+    C_BlockIORequestT *req, uint64_t bytes_cached, uint64_t bytes_dirtied,
+    uint64_t bytes_allocated, uint32_t num_lanes, uint32_t num_log_entries,
```

Comment:
any reason you had to change `uint32_t` to `uint64_t` ?

Reply:
Also, the conditions below in [line#1472](https://github.com/ceph/ceph/blob/master/src/librbd/cache/pwl/AbstractWriteLog.cc#L1472) and [line#1481](https://github.com/ceph/ceph/blob/master/src/librbd/cache/pwl/AbstractWriteLog.cc#L1481) don't make sense for ssd cache, infact `m_free_log_entries` will always be zero for ssd cache - any reason you didn't remove them from here?

Reply:
> Also, the conditions below in [line#1472](https://github.com/ceph/ceph/blob/master/src/librbd/cache/pwl/AbstractWriteLog.cc#L1472) and [line#1481](https://github.com/ceph/ceph/blob/master/src/librbd/cache/pwl/AbstractWriteLog.cc#L1481) don't make sense for ssd cache, infact `m_free_log_entries` will always be zero for ssd cache - any reason you didn't remove them from here?
> 
> Edit: Nevermind - as I recall, this was to reduce code churn at the moment.

It is common code, rwl need it.

Reply:
> any reason you had to change `uint64_t` to `uint32_t`?
> 
> Edit: I suppose this is because these variables are `rwl` specific and the log entries are limited to a million entries

Yes.  In the body these variables are compared with `m_free_lanes`, `m_free_log_entries` and `m_unpublished_reserves` all of which are either `uint32_t` or `unsigned int`.

Reply:
> Also, the conditions below in [line#1472](https://github.com/ceph/ceph/blob/master/src/librbd/cache/pwl/AbstractWriteLog.cc#L1472) and [line#1481](https://github.com/ceph/ceph/blob/master/src/librbd/cache/pwl/AbstractWriteLog.cc#L1481) don't make sense for ssd cache, infact `m_free_log_entries` will always be zero for ssd cache - any reason you didn't remove them from here?

As pointed out by @CongMinYin, this function serves both `rwl` and `ssd` modes.  It wasn't me who designed it this way ;)